libcontainer/configs/validate/validator.go

```diff
@@ -353,3 +355,25 @@ func isHostNetNS(path string) (bool, error) {
 
 	return (st1.Dev == st2.Dev) && (st1.Ino == st2.Ino), nil
 }
+
+// scheduler is to validate scheduler configs according to https://man7.org/linux/man-pages/man2/sched_setattr.2.html
+func scheduler(config *configs.Config) error {
+	s := config.Scheduler
+	if s == nil {
+		return nil
+	}
+	if s.Policy == "" {
+		return fmt.Errorf("scheduler policy is required")
+	}
+	niceValue := s.Nice
+	if niceValue < -20 || niceValue > 19 {
+		return fmt.Errorf("invalid scheduler.nice: %d", niceValue)
```

Comment:
I think it's better to use `s.Nice` directly (drop the intermediate variable) here.

Reply:
I think the original author used an intermediate variable because repeating `config.Scheduler.Nice` three times was too much.

But, just in case an intermediate variable is preferred, it is better to limit its scope:
```go
if n := s.Nice; n < -20 || n > 19 {
	return fmt.Errorf("invalid scheduler.nice: %d", n)
}
```